src/main/java/com/github/cameltooling/lsp/internal/CamelTextDocumentService.java

```diff
@@ -161,15 +164,34 @@ public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completio
 		TextDocumentItem textDocumentItem = openedDocuments.get(uri);
 
 		if (textDocumentItem != null) {
-			if (uri.endsWith(".properties")){
+			if (uri.endsWith(".properties")) {
 				return new CamelPropertiesCompletionProcessor(textDocumentItem, getCamelCatalog(), getCamelKafkaConnectorManager()).getCompletions(completionParams.getPosition(), getSettingsManager(), getKameletsCatalogManager()).thenApply(Either::forLeft);
-			} else if (new CamelKModelineInsertionParser(textDocumentItem).canPutCamelKModeline(completionParams.getPosition())){
-				return new CamelKModelineInsertionProcessor(textDocumentItem).getCompletions().thenApply(Either::forLeft);
+			}
+
+			List<CompletableFuture<List<CompletionItem>>> completions = new LinkedList<>();
+
+			if (new CamelEIPParser(textDocumentItem).canPutEIP(completionParams.getPosition())){
+				completions.add(new CamelEIPParser(textDocumentItem).getCompletions());
+			}
+
+			if (new CamelKModelineInsertionParser(textDocumentItem).canPutCamelKModeline(completionParams.getPosition())){
+				completions.add(new CamelKModelineInsertionProcessor(textDocumentItem).getCompletions());
 			} else if (new CamelKModelineParser().isOnCamelKModeline(completionParams.getPosition().getLine(), textDocumentItem)){
-				return new CamelKModelineCompletionprocessor(textDocumentItem, getCamelCatalog()).getCompletions(completionParams.getPosition()).thenApply(Either::forLeft);
+				completions.add(new CamelKModelineCompletionprocessor(textDocumentItem, getCamelCatalog()).getCompletions(completionParams.getPosition()));
 			} else {
-				return new CamelEndpointCompletionProcessor(textDocumentItem, getCamelCatalog(), getKameletsCatalogManager()).getCompletions(completionParams.getPosition(), getSettingsManager()).thenApply(Either::forLeft);
+				completions.add(new CamelEndpointCompletionProcessor(textDocumentItem, getCamelCatalog(), getKameletsCatalogManager()).getCompletions(completionParams.getPosition(), getSettingsManager()));
 			}
+
+			//Combining all completable futures completion lists
+			//Source: https://nurkiewicz.com/2013/05/java-8-completablefuture-in-action.html
+			return CompletableFuture
+					.allOf(completions.toArray(new CompletableFuture[completions.size()]))
+					.thenApply(
+							v -> completions.stream()
+									.map(future -> future.join())
+									.flatMap(x -> x.stream())
+									.collect(Collectors.toList())
+					).thenApply(Either::forLeft);
```

Comment:
This is needed because we want to be able to add different completions that have different triggers